### ISSUE TITLE
Consider exposing ITerminalInstance.shellLaunchConfig as a promise

### ISSUE THREAD

**Issue by Tyriar** [microsoft/vscode#90993]
This could avoid problems like this https://github.com/microsoft/vscode/pull/90847

**Comment by dormesica** [comment]
@Tyriar To solve #90415 I initially wanted to do something like this but it felt like a bigger change so I went with `await instace.processReady`.

If promise is an acceptable solution and that's OK with you, I would like to help with that.

**Comment by Tyriar** [comment]
@dormesica you can give it a shot, I mentioned consider as I'm not 100% sure whether we want to do this just yet or not. It does seems like what we should be doing based on what we saw with #90847 though.

**Comment by joaomoreno** [comment]
@Tyriar Please remember to create scoped branches like `tyriar/fix-90993` instead of `#90993`.

**Comment by Tyriar** [comment]
@joaomoreno yep,this was a one off because I hardly ever push branches to origin from remotes and didn't rename it
